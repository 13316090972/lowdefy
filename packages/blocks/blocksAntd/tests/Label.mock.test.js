/*
  Copyright 2020-2021 Lowdefy, Inc

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
*/

import { runMockRenderTests } from '@lowdefy/block-tools';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Row, Col } from 'antd';

Enzyme.configure({ adapter: new Adapter() });
import Block from '../src/blocks/Label/Label';
import examples from '../demo/examples/Label.yaml';
import meta from '../src/blocks/Label/Label.json';

jest.mock('antd/lib/row', () => {
  const comp = jest.fn(() => 'mocked');
  return comp;
});

jest.mock('antd/lib/col', () => {
  const comp = jest.fn(() => 'mocked');
  return comp;
});

const mocks = [
  {
    name: 'Row',
    fn: Row,
  },
  {
    name: 'Col',
    fn: Col,
  },
];

runMockRenderTests({ examples, Block, meta, mocks, enzyme: { mount } });
