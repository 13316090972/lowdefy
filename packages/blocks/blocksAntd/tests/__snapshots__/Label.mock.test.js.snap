// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mock render - default - value[0] - Col 1`] = `Array []`;

exports[`Mock render - default - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="default_input"
            title="default"
          >
            default
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "default",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.align: left - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.align: left - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.align: left_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.align: left",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.align: right - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.align: right - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.align: right_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.align: right",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.colon: false - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.colon: false - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="ant-form-item-no-colon {\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.colon: false_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.colon: false",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.colon: true - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.colon: true - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.colon: true_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.colon: true",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.disabled: true - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.disabled: true - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        false,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.disabled: true",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.extra: Extra long - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.extra: Extra long - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.extra: Extra long_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
          <div
            className="ant-form-item-explain ant-form-item-extra {\\"style\\":[{\\"marginTop\\":0},null]}"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commod

          </div>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.extra: Extra long",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.extra: Extra long with inline = true - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.extra: Extra long with inline = true - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"overflow\\":\\"inherit\\",\\"whiteSpace\\":false,\\"marginBottom\\":8,\\"height\\":30}}"
          flex="0 1 auto"
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.extra: Extra long with inline = true_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          flex="1 1 auto"
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
          <div
            className="ant-form-item-explain ant-form-item-extra {\\"style\\":[{\\"marginTop\\":0},null]}"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commod

          </div>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{\\"flexWrap\\":\\"inherit\\"}}",
      "id": "properties.extra: Extra long with inline = true",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.extra: Extra long with inline = true, size = large - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.extra: Extra long with inline = true, size = large - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"overflow\\":\\"inherit\\",\\"whiteSpace\\":false,\\"marginBottom\\":8,\\"height\\":\\"40\\"}}"
          flex="0 1 auto"
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.extra: Extra long with inline = true, size = large_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          flex="1 1 auto"
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
          <div
            className="ant-form-item-explain ant-form-item-extra {\\"style\\":[{\\"marginTop\\":0},null]}"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commod

          </div>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{\\"flexWrap\\":\\"inherit\\"}}",
      "id": "properties.extra: Extra long with inline = true, size = large",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.extra: Extra long with inline = true, size = small - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.extra: Extra long with inline = true, size = small - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"overflow\\":\\"inherit\\",\\"whiteSpace\\":false,\\"marginBottom\\":0,\\"height\\":22}}"
          flex="0 1 auto"
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.extra: Extra long with inline = true, size = small_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          flex="1 1 auto"
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
          <div
            className="ant-form-item-explain ant-form-item-extra {\\"style\\":[{\\"marginTop\\":-4},null]}"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commod

          </div>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{\\"flexWrap\\":\\"inherit\\"}}",
      "id": "properties.extra: Extra long with inline = true, size = small",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.extra: Extra long with inline true - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.extra: Extra long with inline true - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"overflow\\":\\"inherit\\",\\"whiteSpace\\":false,\\"marginBottom\\":8,\\"height\\":30}}"
          flex="0 1 auto"
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.extra: Extra long with inline true_input"
            title="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commod
"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commod

          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          flex="1 1 auto"
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{\\"flexWrap\\":\\"inherit\\"}}",
      "id": "properties.extra: Extra long with inline true",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.extra: Extra text - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.extra: Extra text - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.extra: Extra text_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
          <div
            className="ant-form-item-explain ant-form-item-extra {\\"style\\":[{\\"marginTop\\":0},null]}"
          >
            Extra text
          </div>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.extra: Extra text",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.extraStyle: border 1px purple - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.extraStyle: border 1px purple - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.extraStyle: border 1px purple_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
          <div
            className="ant-form-item-explain ant-form-item-extra {\\"style\\":[{\\"marginTop\\":0},{\\"border\\":\\"1px solid purple\\"}]}"
          >
            Extra text
          </div>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.extraStyle: border 1px purple",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.feedbackStyle: border 1px blue - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.feedbackStyle: border 1px blue - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.feedbackStyle: border 1px blue_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
          <div
            className="ant-form-item-explain ant-form-item-extra {\\"style\\":[{\\"marginTop\\":0},null]}"
          >
            The extra content
          </div>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.feedbackStyle: border 1px blue",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.inline: true - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.inline: true - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"overflow\\":\\"inherit\\",\\"whiteSpace\\":false,\\"marginBottom\\":8,\\"height\\":30}}"
          flex="0 1 auto"
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.inline: true_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          flex="1 1 auto"
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{\\"flexWrap\\":\\"inherit\\"}}",
      "id": "properties.inline: true",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.span: 2 - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.span: 2 - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          md={
            Object {
              "span": 2,
            }
          }
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.span: 2_input"
            title="This is the long label title wrapped by short span"
          >
            This is the long label title wrapped by short span
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          md={
            Object {
              "span": 22,
            }
          }
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.span: 2",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.span: 2 with long work title - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.span: 2 with long work title - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          md={
            Object {
              "span": 2,
            }
          }
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.span: 2 with long work title_input"
            title="Thisisthelonglabeltitlea"
          >
            Thisisthelonglabeltitlea
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          md={
            Object {
              "span": 22,
            }
          }
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.span: 2 with long work title",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.title - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.title - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.title_input"
            title="Label title"
          >
            Label title
          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.title",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;

exports[`Mock render - properties.title long - value[0] - Col 1`] = `Array []`;

exports[`Mock render - properties.title long - value[0] - Row 1`] = `
Array [
  Array [
    Object {
      "children": Array [
        <mockConstructor
          className="ant-form-item-label ant-form-item-label-left {\\"style\\":{\\"whiteSpace\\":\\"normal\\",\\"marginBottom\\":8,\\"height\\":30}}"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <label
            className="{\\"style\\":[{\\"height\\":\\"100% !important\\"},null]}"
            htmlFor="properties.title long_input"
            title="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commod
"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commod

          </label>
        </mockConstructor>,
        <mockConstructor
          className="ant-form-item-control"
          sm={
            Object {
              "span": 24,
            }
          }
          xs={
            Object {
              "span": 24,
            }
          }
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                style={
                  Object {
                    "border": "1px solid red",
                    "padding": 10,
                  }
                }
              >
                content
              </div>
            </div>
          </div>
          <ForwardRef
            motionAppear={true}
            motionName="show-help"
            removeOnLeave={true}
            visible={false}
          >
            [Function]
          </ForwardRef>
        </mockConstructor>,
      ],
      "className": "ant-form-item {\\"style\\":{}}",
      "id": "properties.title long",
      "style": Object {
        "marginBottom": 0,
      },
    },
    Object {},
  ],
]
`;
